optuna/structs.py

```diff
@@ -96,7 +93,81 @@ class FrozenTrial(
             :class:`~optuna.study.Study.study_id` to identify a :class:`~optuna.trial.Trial`.
     """
 
-    internal_fields = ['distributions', 'trial_id']
+    def __init__(
+        self,
+        number,  # type: int
+        state,  # type: TrialState
+        value,  # type: Optional[float]
+        datetime_start,  # type: Optional[datetime]
+        datetime_complete,  # type: Optional[datetime]
+        params,  # type: Dict[str, Any]
+        distributions,  # type: Dict[str, BaseDistribution]
+        user_attrs,  # type: Dict[str, Any]
+        system_attrs,  # type: Dict[str, Any]
+        intermediate_values,  # type: Dict[int, float]
+        trial_id,  # type: int
+    ):
+        self.number = number
+        self.state = state
+        self.value = value
+        self.datetime_start = datetime_start
+        self.datetime_complete = datetime_complete
+        self.params = params
+        self.distributions = distributions
+        self.user_attrs = user_attrs
+        self.system_attrs = system_attrs
+        self.intermediate_values = intermediate_values
+        self._trial_id = trial_id
+
+    internal_fields = ['distributions', '_trial_id']
+    _fields = [
+        'number', 'state', 'value', 'datetime_start', 'datetime_complete', 'params',
+        'distributions', 'user_attrs', 'system_attrs', 'intermediate_values', '_trial_id']
+
+    def __eq__(self, other):
+        # type: (Any) -> bool
+
+        if not isinstance(other, type(self)):
+            return False
+        return all(getattr(self, field) == getattr(other, field)
+                   for field in self._fields)
+
+    def __ne__(self, other):
+        # type: (Any) -> bool
+
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        # type: () -> int
+
+        return hash(tuple(getattr(self, field) for field in self._fields))
+
+    def __repr__(self):
+        # type: () -> str
+
+        return ('{cls}({kwargs})'.format(
+            cls=self.__class__.__name__,
+            kwargs=', '.join('{field}={value}'.format(
+                field=field if not field.startswith('_') else field[1:],
+                value=repr(getattr(self, field))) for field in self._fields)))
+
+    @property
+    def trial_id(self):
+        # type: () -> int
+        """Return trial ID.
+
+        Note that the use of this is deprecated.
+        Please use :attr:`~optuna.trial.FrozenTrial.number` instead.
+
+        Returns:
+            A trial ID.
+        """
+
+        warnings.warn(
+            'The use of `FrozenTrial.trial_id` is deprecated. '
+            'Please use `FrozenTrial.number` instead.', DeprecationWarning)
+
```

Comment:
Because `warnings.warn(..., DeprecationWarning)` doesn't print messages to terminal by default, `Trial.trial_id` uses Optuna logger to output log messages in addition to `warnings.warn()` (see: [trial.py#L521](https://github.com/pfnet/optuna/blob/a9bacebc416a2acdf6924a46ab8ff46378afc275/optuna/trial.py#L521)) .

Reply:
Sorry I missed that. Aligned to `Trial.trial_id` as you suggested.